### ISSUE TITLE
Terragrunt destroy takes very long becuase of git show-toplevel

### ISSUE THREAD

**Issue by dustyhorizon** [gruntwork-io/terragrunt#2511]
Hello,

I have a monorepo with several terragrunt deployments. When I want to destroy a specific terragrunt resource via `terragrunt destroy`, terragrunt takes a very long time to resolve for affected dependencies to destroy because it crawls for affected dependencies via `git rev-parse --show-toplevel`. This makes a simple resource destruction take ~10mins which is a really long time.

Is there a smarter way to do this? Maybe by specifying a boundary of which to crawl for dependencies?

**Comment by pbnsh** [comment]
We also affected by `terragrunt destroy` takes long time issue, `terragrunt apply` is not affected and is near instant.

terragrunt version 0.51.6

When running `terragrunt destroy --terragrunt-log-level debug` we can see that
terragrunt goes through every child directory in the repo and evaluates locals, which takes time.


We are not using dependencies much, and as a workardound, we've done the following:

```
mkdir $REPO_ROOT/0-terragrunt-destroy-workaround
tee $REPO_ROOT/0-terragrunt-destroy-workaround/terragrunt.hcl <<EOF >/dev/null
locals {
  raise = get_env("DOES_NOT_EXIST")
}
EOF
```
The workaround causes terragrunt to fail early the "evaluate all locals" stage, but
doesn't seem to affect destroy/apply commands and it's working fine from what we can tell.


Why terragrunt tries to evaluate all locals in every child directory when doing destroy even when no dependencies are used?

**Comment by n00borama** [comment]
Anyone know which version introduced this bug?
We've just moved to terraform 1.6.2 and terragrunt 0.52.5 and terragrunt destroy actions now take 30mins or more.

We use a mono-repo for a 'controlplane layer' and having a destroy action evaluate everything takes an age.

Plan and Apply don't have this behaviour, so it seems strange that Destroy does.

**Comment by propyless** [comment]
Hitting the same issue, was updating our binaries and noticed that it was taking an excruciatingg amount of time. Even on the latest version so this is not fixed yet.. so will lock our version to 0.51.8 as well, which is working fine.

@denis256 this seems like a bit of a bug, any way to raise priority on this ?

**Comment by gercograndia** [comment]
Same issue here, destroying takes about 15 minutes before it even starts showing something. v0.52.5, not the most recent but I understand it isn't solved in the latest version anyways.

**Comment by circa10a** [comment]
This issue is still present. It seems this functionality is running git rev-parse even if there are no .git directories which makes for ugly output and there's no way to suppress it even though it doesn't functionally affect terraform:

```
fatal: not a git repository (or any of the parent directories): .git

exit status 128
```

**Comment by kbcz1989** [comment]
Hello everyone,

this should be better now since #3534 and release [v0.68.8](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.68.8)

try running:
`terragrunt destroy --terragrunt-no-destroy-dependencies-check`

**Comment by yhakbar** [comment]
@dustyhorizon ,

Would you be able to confirm this? If so, we can close out this issue.

**Comment by gercograndia** [comment]
Works for me. Thanks guys!

**Comment by yhakbar** [comment]
Thanks for confirmation, @gercograndia .

Closing this issue out. Please feel free to open another if you have any other issues, @dustyhorizon !

**Comment by andrea-maruccia-neterium** [comment]
TLDR: `terragrunt destroy --no-destroy-dependencies-check`
